srcjs/shinyapp.js

```diff
@@ -375,6 +375,18 @@ var ShinyApp = function() {
     }
   };
 
+
+  // Narrows a scopeComponent -- an input or output object -- to one constrained
+  // by nsPrefix. Returns a new object with keys removed and renamed as
+  // necessary.
+  function narrowScope(scopeComponent, nsPrefix) {
+    return mapKeys(pickBy(scopeComponent, function(val, key) {
```

Comment:
1. Feel free to use arrow functions
2. I found the indentation a little deceptive somehow, i.e. hard to tell whether the second closure belongs to mapKeys or pickBy
3. Can you make it so the lodash-alike functions are actually accessed via a `_` object? So we can swap them out in the future and also to make it obvious that the semantics of these functions must not be changed (unless it's to match lodash even more closely).